File: Java/1814. Count Nice Pairs in an Array.java
class Solution {
    public int countNicePairs(int[] nums) {
        HashMap<Integer, Integer> mp= new HashMap<>();
        int res = 0;
        for(int i : nums){
            if(mp.containsKey(i - reverse(i))){
                res = (res + mp.get(i - reverse(i))) % 1000000007;
            }
            mp.put(i - reverse(i),mp.getOrDefault(i - reverse(i),0)+1);
        }
        return res;
    }
    public int reverse(int i){
        long rev=0;
        while(i!=0){
            rev=rev*10 + i%10;
            i/=10;
        }
        return (int)rev;
    }
}

File: JavaScript/1814. Count Nice Pairs in an Array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var countNicePairs = function(nums) {
    var mp = new Map();
    var res = 0;
    for(var i of nums){
        if(i - reverse(i) in mp){
            res = (res + mp[i - reverse(i)]) % 1000000007;
            mp[i - reverse(i)]++;
        }
        else{
            mp[i - reverse(i)] = 1;
        }
    }
    return res;
};
var reverse = function(i){
    var rev=0;
    while(i!=0){
        rev=rev*10 + i%10;
        i=Math.floor(i/10);
    }
    return rev;
};
